test/jdk/java/security/cert/CertPathValidator/crlDP/CheckAllCRLs.java

```diff
@@ -0,0 +1,217 @@
+import java.security.KeyPairGenerator;
+import java.security.PublicKey;
+import java.security.cert.CertificateFactory;
+import java.security.cert.CertPath;
+import java.security.cert.CertPathValidator;
+import java.security.cert.CertPathValidatorException;
+import java.security.cert.PKIXParameters;
+import java.security.cert.TrustAnchor;
+import java.security.cert.X509Certificate;
+import java.security.cert.X509CRL;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import sun.security.x509.AuthorityKeyIdentifierExtension;
+import sun.security.x509.CRLDistributionPointsExtension;
+import sun.security.x509.CRLExtensions;
+import sun.security.x509.CRLNumberExtension;
+import sun.security.x509.DistributionPoint;
+import sun.security.x509.Extension;
+import sun.security.x509.GeneralName;
+import sun.security.x509.GeneralNames;
+import sun.security.x509.KeyIdentifier;
+import sun.security.x509.URIName;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CRLImpl;
+import static sun.security.x509.X509CRLImpl.TBSCertList;
+import sun.security.testlibrary.CertificateBuilder;
+
+/*
+ * @test
+ * @bug 8200566
+ * @summary Check that CRL validation continues to check other CRLs in
+ *          CRLDP extension after CRL fetching errors and exhibits same
+ *          behavior whether CRL cache is fresh or stale.
+ * @modules java.base/sun.security.x509
+ *          java.base/sun.security.util
+ * @library ../../../../../java/security/testlibrary
+ * @build CertificateBuilder CheckAllCRLs
+ * @run main/othervm -Dcom.sun.security.enableCRLDP=true CheckAllCRLs
+ */
+public class CheckAllCRLs {
+
+    public static void main(String[] args) throws Exception {
+
+        CertificateBuilder cb = new CertificateBuilder();
+
+        // Create CA cert
+        KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
+        KeyPair rootKeyPair = kpg.genKeyPair();
+        X509Certificate rootCert = createCert(cb, "CN=Root CA",
+            rootKeyPair, rootKeyPair, null, "SHA384withRSA", true, false);
+
+        // Create EE cert. This EE cert will contain a CRL Distribution
+        // Points extension with two DistributionPoints - one will be a HTTP
+        // URL to a non-existant HTTP server, and the other will be a File
+        // URL to a file containing the CRL.
+        KeyPair eeKeyPair = kpg.genKeyPair();
+        X509Certificate eeCert1 = createCert(cb, "CN=End Entity",
+            rootKeyPair, eeKeyPair, rootCert, "SHA384withRSA", false, true);
+
+        // Create another EE cert. This EE cert is similar in that it contains
+        // a CRL Distribution Points extension but with one DistributionPoint
+        // containing 2 GeneralName URLs as above.
+        X509Certificate eeCert2 = createCert(cb, "CN=End Entity",
+            rootKeyPair, eeKeyPair, rootCert, "SHA384withRSA", false, false);
+
+        // Create a CRL with no revoked certificates and store it in a file
```

Comment:
I think the test is based on a fact that if both paths (HTTP and file) fail then validation would fail because there is no way to check for revocation. However, I have a slightest concern that what if it does not fail and everything goes on and validation succeeds. So, if the CRL is not empty and the test detects the cert is revoked it will be more reliable.

Reply:
Yes, this is a good suggestion, as something could go undetected later on. I will update the CRL so that the certificate is revoked and then the test should always expect a failure with the proper reason.